### ISSUE TITLE
Wrong string extracted, ttag adds an underscore at the beginning of a variable.

### ISSUE THREAD

**Issue by dhoko** [ttag-org/ttag-cli#99]
:wave: 

I got a question from a translator about a variable with an underscore at the beginning of a variable. Issue is, I can't find it inside our source file.
After a few tests, I was able to isolate the bug, I have weird behavior, where the length of the file seems to fix the issue (_here only one line is enough_).

## Broken output ttag

```js
import { c, msgid } from 'ttag';

const createMoveLinksNotifications = (
    toMove: FileBrowserItem[],
    {
        moved,
        failed
    }: {
        moved: {
            Name: string;
            Type: LinkType;
        }[];
        failed: string[];
    }
) => {
    const movedLinksCount = moved.length;
    const failedMovesCount = failed.length;

    if (movedLinksCount) {
        const [{ Name: firstItemName }] = moved;
        const notificationMessages = {
            allFiles: c('Notification').ngettext(
                msgid`"${firstItemName}" successfully moved`,
                `${movedLinksCount} files successfully moved`,
                movedLinksCount
            ),
            allFolders: c('Notification').ngettext(
                msgid`"${firstItemName}" successfully moved`,
                `${movedLinksCount} folders successfully moved`,
                movedLinksCount
            ),
            mixed: c('Notification').ngettext(
                msgid`"${firstItemName}" successfully moved`,
                `${movedLinksCount} items successfully moved`,
                movedLinksCount
            )
        };
        const notificationMessage = getNotificationTextForItemList(
            moved.map((item) => item.Type),
            notificationMessages
        );

        createNotification({
            type: 'success',
            text: notificationMessage
        });
    }

    if (!failedMovesCount) {
        return;
    }

    const first = toMove.find(({ LinkID }) => LinkID === failed[0]);
    const firstItemName = first.Name;

};
```

file.pot

```pot
msgid ""
msgstr ""
"Content-Type: text/plain; charset=utf-8\n"
"Plural-Forms: nplurals=2; plural=(n!=1);\n"

#: test.js:22
#: test.js:27
#: test.js:32
#, javascript-format
msgctxt "Notification"
msgid "\"${ _firstItemName }\" successfully moved"
msgid_plural "${ movedLinksCount } files successfully moved"
msgstr[0] ""
msgstr[1] ""
```

## Correct output from ttag

```js
import { c, msgid } from 'ttag';

const createMoveLinksNotifications = (
    toMove: FileBrowserItem[],
    {
        moved,
        failed
    }: {
        moved: {
            Name: string;
            Type: LinkType;
        }[];
        failed: string[];
    }
) => {
    const movedLinksCount = moved.length;
    const failedMovesCount = failed.length;

    if (movedLinksCount) {
        const [{ Name: firstItemName }] = moved;
        const notificationMessages = {
            allFiles: c('Notification').ngettext(
                msgid`"${firstItemName}" successfully moved`,
                `${movedLinksCount} files successfully moved`,
                movedLinksCount
            ),
            allFolders: c('Notification').ngettext(
                msgid`"${firstItemName}" successfully moved`,
                `${movedLinksCount} folders successfully moved`,
                movedLinksCount
            ),
            mixed: c('Notification').ngettext(
                msgid`"${firstItemName}" successfully moved`,
                `${movedLinksCount} items successfully moved`,
                movedLinksCount
            )
        };
        const notificationMessage = getNotificationTextForItemList(
            moved.map((item) => item.Type),
            notificationMessages
        );

        createNotification({
            type: 'success',
            text: notificationMessage
        });
    }

    if (!failedMovesCount) {
        return;
    }

    const first = toMove.find(({ LinkID }) => LinkID === failed[0]);
    // const firstItemName = first.Name;

};
``` 

file.pot
```pot
msgid ""
msgstr ""
"Content-Type: text/plain; charset=utf-8\n"
"Plural-Forms: nplurals=2; plural=(n!=1);\n"

#: test.js:22
#: test.js:27
#: test.js:32
#, javascript-format
msgctxt "Notification"
msgid "\"${ firstItemName }\" successfully moved"
msgid_plural "${ movedLinksCount } files successfully moved"
msgstr[0] ""
msgstr[1] ""
```

the only difference inside the js source is I commented the line 
```pot
    // const firstItemName = first.Name;
```

## About ttag
- version ttag 1.7.22
- version ttag-cli 1.8.3                                       

command
```sh
$ npx ttag extract test.js  -o test.pot
```

> The original file is a `.tsx`

**Comment by dhoko** [comment]
:wave:  @AlexMost It's a scope issue.

### Broken
```js
// File: test.js
if (true) {
  const name = 'monique';
  return c('ctx').t`Hey ${name}`
}

if (false) {
  const name = 'jeanne';
  return c('ctx').t`Hey ${name}`
}
```
```sh
npx ttag extract test.js && cat translations.pot
```

It gives us a broken string `msgid "Hey ${ _name }"

### Fix

```js
// File: test.js
if (true) {
  const name = 'monique';
  return c('ctx').t`Hey ${name}`
}

if (false) {
  return (() => {
    const name = 'jeanne';
    return c('ctx').t`Hey ${name}`
  })()
}
```
```sh
npx ttag extract test.js && cat translations.pot
```

It gives us the expected string `msgid "Hey ${ name }"

**Comment by AlexMost** [comment]
Hi @dhoko! Thanks for the research. I guess it's something related to how babel transpiles js and how it deals with the scopes need more investigation on this question

**Comment by kevinlul** [comment]
Was this covered by #136?

**Comment by AlexMost** [comment]
seems like this one should be fixed in `ttag-cli@1.10.2`. Keep in mind, that you can lose those translations in `.po` that had that scope issue previously.
